src/data-types/money.js

```diff
@@ -23,14 +23,22 @@ module.exports = {
     }
   },
 
-  validate: function(value) {
-    if (value == null) {
+  validate(value) {
+    if (value === undefined || value === null) {
       return null;
     }
-    value = parseFloat(value);
-    if (isNaN(value)) {
-      return new TypeError('Invalid number.');
+
+    let numberValue;
+    if (typeof value === 'number') {
+      numberValue = value;
+    } else {
+      numberValue = parseFloat(value);
     }
-    return value;
+
+    if (!Number.isFinite(numberValue) || (typeof value === 'string' && value !== numberValue.toString()) || numberValue.toString() !== numberValue.toFixed(4)) {
+      return new TypeError(`The given value could not be converted to ${this.name}`);
```

Comment:
How about removing or having an alternative for `numberValue.toString() !== numberValue.toFixed(4)`? 

This condition is too strict, if input has scale < 4 it will throw exception, but driver should allow scale <= 4

Reply:
Can you also add a test for money/smallmoney type with scale < 4? All current tests are for scale 4.

Reply:
You're right, I missed handling of numbers that have less than 4 numbers after the decimal point. 😓

Reply:
`numberValue.toString().split('.')[1].length <= 4`